command/weather.go

```diff
@@ -0,0 +1,186 @@
+package command
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/innogames/slack-bot/bot"
+	"github.com/innogames/slack-bot/bot/config"
+	"github.com/innogames/slack-bot/bot/matcher"
+	"github.com/innogames/slack-bot/client"
+	"github.com/nlopes/slack"
+	"log"
+	"net/http"
+	"time"
+)
+
+func NewWeatherCommand(slackClient client.SlackClient, config config.OpenWeather) bot.Command {
+	return WeatherCommand{slackClient, config}
+}
+
+type City struct {
+	Id   int    `json:"id"`
+	Name string `json:"name"`
+}
+
+type Coord struct {
+	Lon float64 `json:"lon"`
+	Lat float64 `json:"lat"`
+}
+
+type Weather struct {
+	Id          int    `json:"id"`
+	Main        string `json:"main"`
+	Description string `json:"description"`
+	Icon        string `json:"icon"`
+}
+
+type Wind struct {
+	Speed float64 `json:"speed"`
+	Deg   float64 `json:"deg"`
+}
+
+type Clouds struct {
+	All int `json:"all"`
+}
+
+type Rain struct {
+	Threehr int `json:"3h"`
+}
+
+type Main struct {
+	Temp     float64 `json:"temp"`
+	Pressure int     `json:"pressure"`
+	Humidity int     `json:"humidity"`
+	Temp_min float64 `json:"temp_min"`
+	Temp_max float64 `json:"temp_max"`
+}
+
+type Sys struct {
+	Country string `json:"country"`
+	Sunrise int    `json:"sunrise"`
+	Sunset  int    `json:"sunset"`
+}
+
+type CurrentWeatherResponse struct {
+	Coord   Coord     `json:"coord"`
+	Weather []Weather `json:"weather"`
+	Main    Main      `json:"main"`
+	Wind    Wind      `json:"wind"`
+	Rain    Rain      `json:"rain"`
+	Clouds  Clouds    `json:"clouds"`
+	Sys     Sys       `json:"sys"`
+	Dt      int       `json:"dt"`
+	Id      int       `json:"id"`
+	Name    string    `json:"name"`
+}
+
+type ForecastResponse struct {
+	City    `json:"city"`
+	Coord   `json:"coord"`
+	Country string `json:"country"`
+	List    []struct {
+		Dt      int `json:"dt"`
+		Main    `json:"main"`
+		Weather `json:"weather"`
+		Clouds  `json:"clouds"`
+		Wind    `json:"wind"`
+	} `json:"list"`
+}
+
+type WeatherCommand struct {
+	slackClient client.SlackClient
+	config      config.OpenWeather
+}
+
+func (c WeatherCommand) GetMatcher() matcher.Matcher {
+	return matcher.NewRegexpMatcher("weather", c.GetWeather)
+}
+
+func (c WeatherCommand) GetWeather(match matcher.Result, event slack.MessageEvent) {
+	url := fmt.Sprintf("%s?q=%s&units=%s&appid=%s", c.config.Url, c.config.Location, c.config.Units, c.config.Apikey)
+	response, err := http.Get(url)
+	if err != nil {
+		c.slackClient.Reply(event, "Api call returned an err: "+err.Error())
+	}
+
+	var record CurrentWeatherResponse
+	err = json.NewDecoder(response.Body).Decode(&record)
+	if err != nil {
+		log.Println(err)
```

Comment:
use `c.slackClient.ReplyError(event, err)` to log the error properly + respond it to the user